src/Data/DataResponse.php

```diff
@@ -0,0 +1,165 @@
+            return $this->dataStream = $this->response->getBody();
+        }
+
+        throw new \RuntimeException('Data must be a string value.');
+    }
+
+    public function getHeader($name): array
+    {
+        return $this->response->getHeader($name);
+    }
+
+    public function getHeaderLine($name): string
+    {
+        return $this->response->getHeaderLine($name);
+    }
+
+    public function getHeaders(): array
+    {
+        return $this->response->getHeaders();
+    }
+
+    public function getProtocolVersion(): string
+    {
+        return $this->response->getProtocolVersion();
+    }
+
+    public function getReasonPhrase(): string
+    {
+        return $this->response->getReasonPhrase();
+    }
+
+    public function getStatusCode(): int
+    {
+        return $this->response->getStatusCode();
+    }
+
+    public function hasHeader($name): bool
+    {
+        return $this->response->hasHeader($name);
+    }
+
+    public function withAddedHeader($name, $value): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withAddedHeader($name, $value);
+        return $response;
+    }
+
+    public function withBody(StreamInterface $body): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withBody($body);
+        $response->dataStream = $body;
+        return $response;
+    }
+
+    public function withHeader($name, $value): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withHeader($name, $value);
+        return $response;
+    }
+
+    public function withoutHeader($name): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withoutHeader($name);
+        return $response;
+    }
+
+    public function withProtocolVersion($version): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withProtocolVersion($version);
+        return $response;
+    }
+
+    public function withStatus($code, $reasonPhrase = ''): DataResponse
+    {
+        $response = clone $this;
+        $response->response = $this->response->withStatus($code, $reasonPhrase);
+        return $response;
+    }
+
+    public function withResponseFormatter(DataResponseFormatterInterface $responseFormatter): DataResponse
+    {
+        $response = clone $this;
+        $response->responseFormatter = $responseFormatter;
+        return $response;
+    }
+
+    public function withData($data): DataResponse
+    {
+        $response = clone $this;
+        $response->data = $data;
+
+        return $response;
+    }
+
+    public function hasResponseFormatter(): bool
+    {
+        return $this->responseFormatter !== null;
+    }
+
+    public function getResponse(): ResponseInterface
+    {
+        return $this->response;
+    }
+
+    public function getData()
+    {
+        if (is_callable($this->data)) {
```

Comment:
I don't sure that it really needed

Reply:
It is useful for delaying computation. I'd keep it.